### PR TITLE
Add some C++ scripting examples

### PR DESCRIPTION
**Pull request by W2Wizard**
# Add C++ examples for scripting tutorials

As for my continued effort, I tried to add a few C++ examples on how to implement some of the tutorials.
Additionally I took some time to fix a few typos here and there as well as tidy up a picture.

image

### CODE REVIEW

**Comment by stefnotch** [comment]
Apparently DocFx supports something called [tabbed code snippets](https://github.com/dotnet/docfx/blob/920dd053453999acfd93b4d171f55074058a9aff/Documentation/spec/docfx_flavored_markdown.md#section-definition)

I guess this would be a good time to try them out

**Comment by W2Wizard** [comment]
> Apparently DocFx supports something called [tabbed code snippets](https://github.com/dotnet/docfx/blob/920dd053453999acfd93b4d171f55074058a9aff/Documentation/spec/docfx_flavored_markdown.md#section-definition)
> 
> I guess this would be a good time to try them out

Oh very nice! Thanks for pointing that out :D
Seems to work like a charm, I'll see to it, to change the examples to this implemenation instead.
image

**Comment by W2Wizard** [comment]
Well it seems that it sometimes struggles to properly insert the code.

Code:
image

Result:
image

**Comment by stefnotch** [comment]
I think that's a case of spaces vs tabs. I recommend setting your IDE to aggressively autoformat to one style or the other when you hit the save button.

File: manual/scripting/code-examples/scene-objects.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+using FlaxEngine;
+
+public class MyScript : Script
+{
+    public override void OnStart()
+    {
+        // Prints all children names
+        var children = Actor.GetChildren();
```
Comment (mafiesto4) [inline manual/scripting/code-examples/scene-objects.cs:8]: It should be `var children = Actor.Children`

File: manual/scripting/code-examples/scene-objects.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "Engine/Scripting/Script.h"
+
```
Comment (mafiesto4) [inline manual/scripting/code-examples/scene-objects.h:4]: Add `#include "Engine/Debug/DebugLog.h"`

File: manual/scripting/tutorials/change-scene.md [hunk 3]
```diff
@@ -23,19 +25,47 @@ public class SceneChanger : Script
 }
 ```
 
-### 3. Add script
+### C++
+
```
Comment (mafiesto4) [inline manual/scripting/tutorials/change-scene.md:29]: Insert includes:
```cpp
#include "Engine/Input/Input.h"
#include "Engine/Level/Level.h"
#include "Engine/Scripting/Script.h"
```

File: manual/scripting/tutorials/change-scene.md [hunk 4]
```diff
@@ -23,19 +25,47 @@ public class SceneChanger : Script
...
+public:
+    void OnEnable() override;
+    void OnDisable() override;
+    void OnUpdate() override
+    {
+        if (Input::GetKeyDown(KeyboardKeys::G) && AnotherScene.IsValid())
+        {
+			// Does the same as the C# API
+			Level::UnloadAllScenesAsync();
+			Level::LoadSceneAsync(id);
```
Comment (mafiesto4) [inline manual/scripting/tutorials/change-scene.md:50]: Replace `id` with `AnotherScene`

File: manual/scripting/code-examples/scene-objects.cs (hunk 1 above)
Comment (W2Wizard) [reply manual/scripting/code-examples/scene-objects.cs]: Done

File: manual/scripting/code-examples/scene-objects.h (hunk 2 above)
Comment (W2Wizard) [reply manual/scripting/code-examples/scene-objects.h:5]: Done

File: manual/scripting/tutorials/change-scene.md (hunk 3 above)
Comment (W2Wizard) [reply manual/scripting/tutorials/change-scene.md:18]: Done

File: manual/scripting/tutorials/change-scene.md (hunk 4 above)
Comment (W2Wizard) [reply manual/scripting/tutorials/change-scene.md]: Oops!

**Comment by mafiesto4** [comment]
LGTM
except for merge conflict from the previous PR.

**Comment by W2Wizard** [comment]
Mine should overwrite the merge of #58 pretty much, same thing except using code snippet.

**Pull request merged**